src/pip/_internal/req/__init__.py

```diff
@@ -1,10 +1,69 @@
 from __future__ import absolute_import
 
+import logging
+
 from .req_install import InstallRequirement
 from .req_set import RequirementSet
 from .req_file import parse_requirements
+from pip._internal.utils.logging import indent_log
+
 
 __all__ = [
     "RequirementSet", "InstallRequirement",
-    "parse_requirements",
+    "parse_requirements", "install_given_reqs",
 ]
+
+logger = logging.getLogger(__name__)
+
+
+def install_given_reqs(to_install, install_options, global_options=(),
```

Comment:
I'm not too happy I had to do this but I'm pretty sure that this function will move into a nicer place after a refactor of InstallRequirement, which I'll tackle after the Resolver is swapped in with the one from zazo.